addons/mass_mailing/models/mailing.py

```diff
@@ -616,14 +617,19 @@ def action_duplicate(self):
 
     def action_test(self):
         self.ensure_one()
-        ctx = dict(self.env.context, default_mass_mailing_id=self.id, dialog_size='medium')
+        test_wizard_values = {'mass_mailing_id': self.id}
+        test_mails = request.session.get('test_emails')
+        if test_mails:
+            test_wizard_values['email_to'] = test_mails
+        test_mailing_wizard = self.env['mailing.mailing.test'].create(test_wizard_values)
         return {
             'name': _('Test Mailing'),
             'type': 'ir.actions.act_window',
             'view_mode': 'form',
             'res_model': 'mailing.mailing.test',
             'target': 'new',
-            'context': ctx,
+            'context': dict(self.env.context, dialog_size='medium'),
+            'res_id': test_mailing_wizard.id,
         }
```

Comment:
I don't think we need to create the wizard beforehand. We could just pass the context. So something like this:
```python
ctx = dict(self.env.context, default_mass_mailing_id=self.id, dialog_size='medium')
if request.session.get('test_mailing_email_to'):
    ctx.update(default_email_to=request.session["test_mailing_email_to"])
```

Reply:
Agreed